yarn-project/end-to-end/src/e2e_p2p/slashing.test.ts

```diff
@@ -0,0 +1,278 @@
+import { type AztecNodeService } from '@aztec/aztec-node';
+import { sleep } from '@aztec/aztec.js';
+import { RollupAbi, SlashFactoryAbi, SlasherAbi, SlashingProposerAbi } from '@aztec/l1-artifacts';
+
+import fs from 'fs';
+import { getAddress, getContract, parseEventLogs } from 'viem';
+
+import { shouldCollectMetrics } from '../fixtures/fixtures.js';
+import { createNodes } from '../fixtures/setup_p2p_test.js';
+import { AlertChecker, type AlertConfig } from '../quality_of_service/alert_checker.js';
+import { P2PNetworkTest } from './p2p_network.js';
+import { createPXEServiceAndSubmitTransactions } from './shared.js';
+
+const CHECK_ALERTS = process.env.CHECK_ALERTS === 'true';
+
+// Don't set this to a higher value than 9 because each node will use a different L1 publisher account and anvil seeds
+const NUM_NODES = 4;
+const BOOT_NODE_UDP_PORT = 40600;
+
+const DATA_DIR = './data/gossip';
```

Comment:
datadir collides with gossip test

Also probably do not want the qosAlerts as it is not running with those in CI

Reply:
Ah, yes will fix.